file-formats/ddls/ddls.json

```diff
@@ -0,0 +1,126 @@
+
+{ 
+  "$schema": "https://json-schema.org/draft/2020-12/schema", 
+  "$id": "http://sap.com/schema/nrob.json", 
+  "title": "Object Type DDLS", 
+  "description": "DDLS object type", 
+  "type": "object", 
+  "properties": { 
+    "$schema": { 
+      "title": "Schema", 
+      "description": "Format version", 
+      "type": "string" 
+    }, 
+    "header": { 
+      "title": "Header", 
+      "description": "Header", 
+      "type": "object", 
+      "properties": { 
+        "description": { 
+          "title": "Description", 
+          "description": "Description of the ABAP object", 
+          "type": "string", 
+          "maxLength": 60 
+        }, 
+        "originalLanguage": { 
+          "title": "Original Language", 
+          "description": "Original language of the ABAP object", 
+          "type": "string", 
+          "maxLength": 2 
+        }, 
+        "abapLanguageVersion": { 
+          "title": "ABAP Language Version", 
+          "description": "ABAP language version", 
+          "type": "string", 
+          "enum": [ 
+            "standard", 
+            "keyUser", 
+            "cloudDevelopment" 
+          ], 
+          "enumDescriptions": [ 
+            "Standard", 
+            "ABAP for key user extensibility", 
+            "ABAP cloud development" 
+          ], 
+          "default": "standard" 
+        } 
+      }, 
+      "additionalProperties": false, 
+      "required": [ 
+        "description", 
+        "originalLanguage" 
+      ] 
+    }, 
+    "sourceOrigin": { 
+      "title": "Source Origin", 
+      "description": "Source Origin", 
+      "type": "integer", 

```

Comment:
Only type string is possible for enums. I assume that the schema is generated. If so, we have to adapt the generator.

Reply:
If you link the type to ty_source_origin to the corresponding constants like it's done for example in [zif_aff_intf_v1](https://github.com/SAP/abap-file-formats/blob/main/file-formats/intf/type/zif_aff_intf_v1.intf.abap), the generator adds the right type

Reply:
Thanks. Fixed with next commit.

Reply:
> ty_source_origin

Thanks. Will fix.